sdk/client/src/error.rs

```diff
@@ -36,6 +40,16 @@ enum Kind {
     Decode,
     InvalidProof,
     Unknown,
+
+    // Streaming Errors
+    Encode,
+    ConnectionClosed,
+    MessageTooLarge,
+    HttpError,
+    TlsError,
+    IdAlreadyUsed,
+    IdNotFound(Option<StreamJsonRpcResponse>),
+    QueueFullError,
```

Comment:
This is quite a bit to be added just for websocket support. Given streaming clients are completely different from the non-streaming client maybe it makes more sense to have a separate error type?